public/Leo Jayachandran/Glossarizer/libretextsglossarizer.js

```diff
@@ -0,0 +1,629 @@
+                    } else {
+                        exclusions = "";
+                    }
+                    newTerm["term"] = (cols["word"].substring(1).toLowerCase().replace(/<p>/g, " ").replace(/<\/p>/g, " ").trim() + exclusions).trim();
+
+                    //Make Description
+                    if (cols["link"].length) {
+                        let aTagStart = 'href="';
+                        let aTagEnd = '">';
+                        let href = cols["link"].substring(cols["link"].search(aTagStart) + aTagStart.length, cols["link"].search(aTagEnd));
+                        cols["definition"] = `<a href = "${href}" target="_blank">${cols["definition"]}</a>`;
+                    }
+                    if (cols["image"].length) {
+                        cols["definition"] += `<div class='imageContainer'>${cols["image"]}</div>`;
+                    }
+                    if (cols["caption"].length) {
+                        cols["definition"] += `<p class = 'caption'>${cols["caption"]}</p>`;
+                    }
+                    if (cols["source"].length) {
+                        cols["definition"] = cols["definition"].trim() + `<p class = "glossarySource">[Source: ${cols["source"].replace(/<p>/g, " ").replace(/<\/p>/g, " ").trim()}]</p>`;
+                    }
+
+                    newTerm["description"] = cols["definition"];
+
+                    //Add new term
+                    retrievedGlossary.push(newTerm);
+                }
+                return retrievedGlossary;
+            }
+        };
+        let chosenSource = sourceOption || "textbook";
+        retrievedGlossary = await glossaryRetriever[chosenSource]();
+        if (retrievedGlossary.length <= 1) { // Deal with incompatible Glossary
+            return;
+        }
+        let pluginName = 'glossarizer',
+            defaults = {
+                replaceTag: 'div',
+                /* Matching words will be wrapped with abbr tags by default */
+                lookupTagName: 'p, ul, a, div',
+                /* Lookup in either paragraphs or lists. Do not replace in headings */
+                callback: null,
+                /* Callback once all tags are replaced: Call or tooltip or anything you like */
+                replaceOnce: false,
+                /* Replace only once in a TextNode */
+                replaceClass: 'glossarizer_replaced',
+                caseSensitive: false,
+                exactMatch: false
+            };
+        /**
+         * Plugin Name: Glossarizer
+         * Author : Vinay @Pebbleroad
+         * Date: 02/04/2013
+         * Description: Takes glossary terms from a JSON object -> Searches for terms in your html -> Wraps a abbr tag around the matched word
+         * 1. Fixed IE8 bug where whitespace get removed - Had to change `abbr` tag to a block element `div`
+         */
+        /**
+         * Constructor
+         */
+
+        function Glossarizer(el, options) {
```

Comment:
This code appears to be external code from the Glossarizer plugin. Do you really want to embed this here rather than linking to the code externally?

Reply:
I used the original backbone and a lot of it is similar, but I had to adapt it a bit since it acquired the glossary json inside the function. Essentially it's not the same as the original plugin.